packages/openapi-v3/src/__tests__/unit/json-to-schema.unit.ts

```diff
@@ -139,6 +139,31 @@ describe('jsonToSchemaObject', () => {
     expect(jsonToSchemaObject(inputDef)).to.eql(expectedDef);
   });
 
+  it('handles circular references', () => {
+    const schemaJson: JsonSchema = {
+      title: 'ReportState',
+      properties: {
+        states: {type: 'array', items: {$ref: '#/definitions/ReportState'}},
+        benchmarkId: {type: 'string'},
+        color: {type: 'string'},
+      },
+    };
+    schemaJson.definitions = {ReportState: schemaJson};
+    const schema = jsonToSchemaObject(schemaJson);
+    expect(schema).to.eql({
+      title: 'ReportState',
+      properties: {
+        states: {
+          type: 'array',
+          items: {$ref: '#/components/schemas/ReportState'},
+        },
+        benchmarkId: {type: 'string'},
+        color: {type: 'string'},
+      },
+      definitions: {ReportState: {$ref: '#/components/schemas/ReportState'}},
```

Comment:
I find it weird to have a definition that's referencing itself, it's effectively creating a cyclic reference/infinite loop. Can we remove this entry please?